### DISCUSSION TITLE
Unknown alarm codes

### DISCUSSION THREAD

**Discussion by el-dom** [tbnobody/OpenDTU#590] [Q&A] [answered]
I'm getting alarm codes 12 and 47 occasionally on a HM-1200 inverter. https://github.com/tbnobody/OpenDTU/blob/master/lib/Hoymiles/src/parser/AlarmLogParser.cpp does not know them and the troubleshooting list in the user manual also doesn't mention them.
Does anybody know what those codes mean?

![Screenshot 2023-02-15 at 14-19-31 OpenDTU](https://user-images.githubusercontent.com/7406043/219039057-a39b4eff-0e0f-4418-a79c-32e6a70fff74.png)

**Comment by homeautomation2022** [comment, marked as answer]
Ja, ist schon länger bekannt, dass es Unknown Logeinträge gibt (z.B. https://github.com/tbnobody/OpenDTU/issues/121), dazu zählen die Codes: 12, 14, 36, 46, 47
46 und 47 müssen sehr ähnlich sein und kommen manchmal gleichzeitig, aber die Hintergründe sind noch komplett unbekannt. Leider sind diese fehler/Warnungen nirgends bei Hoymiles dokumentiert wurden, aber eventuell über die original DTU herausfindbar.

Vermutlich sind alles unter ID 100 nur Warnungen und scheinen vermutlich von Netzstörungen zu kommen.

**Reply by stefan123t** [reply]
@homeautomation2022 Du kennst Dich doch mit dem Innenleben der Hoymiles Wechselrichter bestens aus.
Was könnte denn diese u.g. FB clamp sein ? 

Event ID | Description
--- | ---
11 | Grid voltage surge
12 | Grid voltage sharp drop
13 | Grid frequency mutation
14 | Grid phase mutation
15 | Grid transient fluctuation
36 | INV overvoltage or overcurrent
46 | FB overvoltage
47 | FB overcurrent
48 | FB clamp overvoltage
49 | FB clamp overvoltage

Die Bedeutung von VW, FW und TW Watch Dog Funktion haben wir unten Dank lemmi bereits identifizieren können ;)

**Comment by el-dom** [comment]
Danke sehr.

**Comment by stefan123t** [comment]
Siehe auch #121

**Comment by lemmi** [comment]
I think ID `73` might indicate a high temperature warning. My `HMT-2250-6T` was running slightly above 80°C when I noticed the entry without a `Stop` time, like this:

Start | Stop | ID | Message
--:| --:| --:|:--
...|...|...|...
12:58:42 | 00:00:00 | 73 | Unknown

After using a blower to cool the thing down to about 65°C, the `Stop` time was replaced with the current time:


Start | Stop | ID | Message
--:| --:| --:|:--
...|...|...|...
12:58:42 | 13:47:45 | 73 | Unknown

I let it heat up again, but it only got to about 79.5°C. No message appeared this time

**Reply by homeautomation2022** [reply]
https://www.hoymiles.com/resources/blog/how-do-hoymiles-4-in-1-microinverters-perform-at-high-temperatures/
OK, der HMT-2250 ist da nicht aufgelistet, aber warum sollte er sich so viel anders verhalten als bei den anderen Modellen, bitte nochmal gegenprüfen mit Beobachtung der Temperatur aus der DTU.

**Reply by lemmi** [reply]
I haven't seen any throttling, so I assume this is just a high temperature warning and nothing else. I'll try to confirm my assumption the next days and try to cool it down a little more slowly to better pinpoint the stop threshold.

**Reply by lemmi** [reply]
So I dumped the `livedata` during the day like this:
```bash
websocat ws://opendtu/livedata | tee -a /var/tmp/$(date -I)_solar
```
Sadly there are no timestamps in the json data and I only noticed that later. 
I cleaned up the json afterwards to list the temperature and the events:
```bash
cat 2023-06-01_solar | jq -c '{temp: .inverters[0].INV."0".Temperature.v, events: .inverters[0].events}'
```

About 120 entries after first hitting 80°C, the event counter went to `2`, which corresponds to another `73` message:
```json
...
{"temp":79.90000153,"events":1}
{"temp":80,"events":1}
... ~120 entries
{"temp":80.30000305,"events":1}
{"temp":80.30000305,"events":1}
{"temp":80.30000305,"events":2}
{"temp":80.30000305,"events":2}
{"temp":80.30000305,"events":2}
... ~670 entries above 80°
{"temp":79.90000153,"events":2}
...
```

So assuming I get roughly one entry per 1s from the livedata endpoint, my best guess is that the high temperature warning appears when the temperature is over 80°C for at least 2 minutes. 

Start | Stop | ID | Message
--:| --:|--:|:--
04:57:40 | 04:57:40 | 1 | Inverter start
14:07:19 | 14:20:40 | 73 | Temperature >80°C

Overall, the event lasted for `13m20s`. That also fits with the 674 events from the livedata (11m14s) + 2 minutes hysteresis.

**Reply by stefan123t** [reply]
@lemmi siehe auch die Liste aus der `assets/warn_code.json` in https://github.com/tbnobody/OpenDTU/issues/121#issuecomment-1752124208
D.h.
71     VW function enable
72     FW function enable
73     TW function enable
Sind die Voltage, Frequency und Temperature Warning/Watch Dog Funktionen ! 
Danke für's ausprobieren und die Erklärung.

**Comment by stefan123t** [comment]
@el-dom Danke für die Meldung, ja es gibt noch ein paar Unbekannte / unknown Fehlermeldungen.
U.a. 
* 12 (https://github.com/tbnobody/OpenDTU/issues/1068),
* 14 (https://github.com/tbnobody/OpenDTU/issues/131),
* 12, 13, 14, 36,  (https://github.com/tbnobody/OpenDTU/issues/121),
* 12, 14, 36, 46, 47, 73 https://github.com/tbnobody/OpenDTU/discussions/590

Ich bin in den `assets/warn_code.json` der Hoymiles S-Miles Installer App fündig geworden. 🥳 

Hier mal die bisher aufgetretenen Fehler oder Warnings aus dem in #121 angehängten warn_code.json (GitHub versteht nur ZIP ?).

11: Grid voltage surge
12: Grid voltage sharp drop
13: Grid frequency mutation
14: Grid phase mutation
36: INV overvoltage or overcurrent
46: FB overvoltage
47: FB overcurrent
71: VW function enable
72: FW function enable
73: TW function enable

Es wird von @tbnobody im Rahmen von #121 in OpenDTU implementiert und daher kann diese Discussion m.E. hiermit beantwortet werden.

**Reply by Tiese** [reply]
Was bedeuten denn die Abkürzungen?

INV steht bestimmt für Inverter, also Wechselrichter.
Aber was bedeutet FB, VW, FW und TW?

Vielen Dank für die Aufklärung ;-)

**Reply by stefan123t** [reply]
Schau mal #121 oder die Diskussion #590 da wird mW bereits dazu gemutmaßt.

VW, FW, TW Voltage, Frequency oder Temperature Warning/WatchDog
FB FlyBack Transformer

**Reply by Tiese** [reply]
Hi stefan123t,
ich danke dir für den Tipp!
Im [#121](https://github.com/tbnobody/OpenDTU/issues/121) steht z.B. für FW - Frequency Watt 
Im [#590](https://github.com/tbnobody/OpenDTU/discussions/590) sind wir doch - da lautet es Watch Dog.

Ist auch egal, ich habe es so abgekürzt in mein Script übernommen.

**Reply by stefan123t** [reply]
Vorschlag von @tbnobody war heute noch Voltage, Frequency / Temperature Warning. Das passt m.E. genauso gut.

**Reply by tbnobody** [reply]
Ich habe einfach mal die chinesischen originaltexte übersetzt. hätte es folgendermaßen implementiert:

```cpp
    { AlarmMessageType_t::ALL, 71, "Grid overvoltage load reduction (VW) function enable", "", "" },
    { AlarmMessageType_t::ALL, 72, "Power grid over-frequency load reduction (FW) function enable", "", "" },
    { AlarmMessageType_t::ALL, 73, "Over-temperature load reduction (TW) function enable", "", "" },
```

**Reply by stefan123t** [reply]
Bitte einheitlich mit over-voltage und entweder Grid oder Power grid bei 71/72 ;) Wobei 73 natürlich keine Grid bezogene over-temperature warning sein kann.

Aber die Frage was genau 46/47 FB overvoltage/overcurrent sind wäre ja noch offen, oder ?

**Comment by Tiese** [comment]
Die Fehler 13, 36, 46 und 47 hatte ich auch "ständig", bis der FI 30mA (worüber der WR läuft) gegen einen 300mA getauscht wurde.
Seitdem ist keiner dieser Fehler mehr aufgetreten!

**Reply by stefan123t** [reply]
Das muß ich mir mal ansehen, danke für den Tipp!
Seitdem ich auf meinen Shelly Plus 1PM eine aktuelle Firmware geflasht habe, war dieses Jahr über auch Ruhe.
Mal sehen ob es jetzt in der kalten Jahreszeit wieder akut wird.

**Reply by Blade138** [reply]
Moin hat sich das Problem erledigt seit dem?

**Comment by gitisgreat2023** [comment]
Was ist error code 53?

Flyback Transformer (FB overvoltage 46, FB overcurrent 47), was passiert da? Geht da was kaputt? Hängt es mit der DC, AC, oder transformers selber zsuammen?

**Comment by Taurus1958** [comment]
Was löst die Evewnt ID 36 eigentlich aus (36: INV overvoltage or overcurrent) bzw. wo passiert hier etwas - am DC Eingang oder am AC Ausgang?
Ich bekomme dieses Event relativ oft. Leite AC von einem HM 800 durch einen HM1500 und danach ins Haus. Die Hoymiles sCloud zeigt keinen Fehler an, der kommt nur bei der OpenDTU.
Wenn ich die Werte kontrolliere, kann ich keine Abweichungen zu den Werten im Datenblatt des HM800 finden. Der HM1500 zeigt im Eventlog der OpenDTU keine derartigen Einträge.
Kann da irgendwas defekt werden oder wie ist diese ID 36 zu bewerten?

**Comment by Taurus1958** [comment]
Hallo, leider NEIN


Gesendet: Montag, 17. März 2025 um 13:53
Von: Blade138 ***@***.***>
An: tbnobody/OpenDTU ***@***.***>
CC: Taurus1958 ***@***.***>,Comment ***@***.***>
Betreff: Re: [tbnobody/OpenDTU] Unknown alarm codes (Discussion #590)

 
Moin hat sich das Problem erledigt seit dem?

**Reply by Blade138** [reply]
Moin, ich hab meine Ursache gefunden. Ich hatte eine Shelly Steckdose direkt neben dem Wechselrichter in der Steckdose. Dort dran waren ein paar Lüfter mit seperatem Netzteil. Immer wenn die Lüfter geschaltet wurden, kam die Meldung. Vielleicht hast du auch nen großen Verbraucher der das verursachen könnte? Kaffeemaschine, Klima, etc?

**Reply by f-1-0** [reply]
Hier ist der alarm code 36 um 7:48 Uhr gemeldet worden, exakt zu dem Zeitpunkt hat die 'große' Ost-West Anlage (Hybrid inverter SMA STP 6.0 SE) mit 2kw+ losgelegt bei klarem Himmel.
